data/persistence/src/commonMain/kotlin/social/androiddev/common/persistence/localstorage/DodoAuthStorageImpl.kt

```diff
@@ -31,29 +33,31 @@ internal class DodoAuthStorageImpl(
      * The user can set up multiple accounts on their device. So we
      * key the AccessToken by the unique server/domain
      */
-    private var diskCache: Map<String, AccessToken>
+    private var diskCache: LinkedHashMap<String, AccessToken>
         get() {
             return settings
                 .getStringOrNull(KEY_ACCESS_TOKENS_CACHE)
                 ?.let { str ->
                     json.decodeFromString(ListSerializer(AccessToken.serializer()), str)
-                        .associateBy { it.server }
+                        .associateByTo(linkedMapOf()) { it.server }
                 }
-                ?: mutableMapOf()
+                ?: linkedMapOf()
```

Comment:
LinkedHashMap is mutable by default?

Reply:
Yes - it extends from MutableMap (on JVM it's typealias to LinkedHashMap, which is also mutable)